### DISCUSSION TITLE
Zooz Water Valve Controller on Ring App not showing

### DISCUSSION THREAD

**Discussion by jamouabbi** [tsightler/ring-mqtt#812] [Support Q&A] [answered]
Hi I have added a Zooz Water Valve controller on my Ring app. I was hoping that the Ringmqtt integration will also show it but it is not. Can you please help

**Comment by tsightler** [comment]
Water value is not included in the list of supported devices so it is not supported.  I do have the discovery data needed to add it, but I just have no time to implement this right now and probably will not any time soon.

**Reply by jamouabbi** [reply]
Is there anyway I can convince you to do it? Maybe buy you a generous cup of coffee?

**Reply by tsightler** [reply, marked as answer]
It's not due to a lack of desire, but a lack of time.  Money can't buy time, well, at least I'm going to guess there's no cup of coffee generous enough!  The overall effort to add it is simple, it's basically just a switch with a different class, but it still takes time, and I just don't have it available right now.  Hopefully in a few months things will free up a little bit, but I've been saying that since October of last year, so who knows.

**Reply by jamouabbi** [reply]
Thank you! I appreciate all you do! let me know if you ever feel like you have time to do it! I would like to contribute a donation to your efforts

**Reply by jamouabbi** [reply]
Any update on the addition of the Zooz water valve?

**Reply by tsightler** [reply]
Over the last few months, what time I've had for this project has been focused almost completely on keeping the plugin working though multiple API changes, with the biggest being the major changes to push API.  The push notifications issue is not really a Ring thing as they just use FCM for push messaging, but the legacy FCM/GCM APIs started shutting down at the end of July and we spent months getting updates into the upstream libraries and then reverse engineering and testing the new push message format, and troubleshooting reliability issues with the updated libraries.

On top of that I've spent untold hours tracking down various issues in other projects on which this project depends, like go2rtc, which had a minor change in how it spawned exec process which cascaded into a major issue.  Also spent tons of time tracing various failure scenarios with live streaming, etc.

All of this work just to keep things working has left little time to add new devices.  I'm soon to release v5.7.0, which will include all of that work above.  If I get really lucky, perhaps I can get water valve added in that version as well, but more likely it will be in a few weeks with a new update.

**Reply by jamouabbi** [reply]
I really appreciate you! Thank you for all your hard work! Do you have a donation page?

**Reply by tsightler** [reply]
Would you be willing to test a dev version to verify functionality?  I should be able to get it added next week if nothing unexpected happens but, as I don't own a valve, it would be good to test it prior to shipping it in production.

**Reply by jamouabbi** [reply]
Of course! I would love to try it out

**Reply by tsightler** [reply]
Ok.  My current plan is to get 5.7.0 out on Sunday evening.  This is important because some users are already seeing failures due to the new style push notifications being used by default in certain cases.  Once that is done, I'll try to get the valve code merged into the dev branch and I'll post here.  Thanks!

**Reply by tsightler** [reply]
@jamouabbi If you are able to test, I have implemented water valve support in the dev branch.  To test the dev branch you simply need to set the "branch" configuration option to "dev" in the addon configuration settings or, for Docker, use the BRANCH environment variable.  Then restart the addon/container and it will pull down the latest version during startup.  The logs should show 5.7.1 in the startup banner.  If you could test and report success/failure, and ideally provide logs if failure, please let me know.

**Reply by jamouabbi** [reply]
I am having a hard time getting the dev version to load. please see logs below
![image](https://github.com/user-attachments/assets/54317a36-a4b7-4f9f-adc9-ca982d115cd7)

**Reply by tsightler** [reply]
Have you already successfully upgraded to the 5.7.0 version of the addon?

**Reply by jamouabbi** [reply]
yes I did

**Reply by tsightler** [reply]
Hmm, very strange, this feature has been around forever and I've never had any issues with it.  Are you restarting or is it restarting on it's own?  Is there definitely plenty of space on your device?

**Reply by jamouabbi** [reply]
Yes I do (see image)
I did it both ways and the same thing happened.
I am wondering if there is another step that I need to take to be able to open the dev version
![image](https://github.com/user-attachments/assets/d4896a1f-048b-476e-baf2-6dedac532c53)

**Reply by tsightler** [reply]
Assuming the container has proper internet access there is nothing else that needs to be done.  It appears that the process is crashing for some reason, maybe a VPN or something that is blocking access to Github?  I honestly have no other suggestions, there's nothing complex going on here, it's just attempting to run `git clone` at this point, so not even any code to troubleshoot.

**Reply by jamouabbi** [reply]
IT WORKED! it worked flawlessly! Thank you so much!!! Do you have a patreon or ko-fi page that supports your work??

**Reply by tsightler** [reply]
Thanks for testing @jamouabbi.  I don't accept donations as I'm afraid it would cause this project to become more like work than a fun distraction as I would feel obligated to work on things in a timely way if people were contributing to me.

**Reply by jamouabbi** [reply]
Can you please reconsider? You are doing a fantastic job and you deserve some monetary appreciation to your work! We all understand that is not a job and that you are doing it for fun and making our lives so much easier in the process. It is merely a token of appreciation for making my life and others so much easier!

**Reply by tsightler** [reply]
I won't change my mind, at least for now, but if you are just itching to contribute, the last question in the general section of the FAQ from the wiki offers some suggestions.

**Reply by jamouabbi** [reply]
I respect that. I want you to know I donated to the National Multiple Sclerosis Society in you honor (I wrote tsightler) and also to the ring-client-api (I also said its for tsightler in the comment).
Thank you again

**Reply by tsightler** [reply]
Thanks, that is really kind of you!
